### ISSUE TITLE
Private URL problem in 4.6.1

### ISSUE THREAD

**Issue by 4tikhonov** [IQSS/dataverse#3739]
Hi guys,

There is something wrong with Private URL functionality as it shows login screen for people trying to access dataset by shared url. It should open private dataset landing page directly.
Can you please take a look?

**Comment by djbrooke** [comment]
Thanks @4tikhonov, I'll take a look.

**Comment by pdurbin** [comment]
@4tikhonov https://demo.dataverse.org is running v. 4.6.1 build 69-1ed82ae. Can you please try to reproduce the bug there?

**Comment by djbrooke** [comment]
Hey @4tikhonov - I tried to reproduce it and it worked as expected on our demo site running 4.6.1. Let me know if you can provide some more specifics.

**Comment by pdurbin** [comment]
@4tikhonov this may be too much information but I wrote a lot of Javadoc on how Private URL works back when I developed it and you're welcome to check it out here if you want: http://phoenix.dataverse.org/site/latest/site/apidocs/edu/harvard/iq/dataverse/privateurl/package-summary.html

**Comment by djbrooke** [comment]
@4tikhonov - I'm going to close this one out. Please feel free to reopen with additional details if you're still experiencing issues.

**Comment by 4tikhonov** [comment]
We're still having this problem. Can you please share some private url page from demo.dataverse.org so I can compare if it works the same?

**Comment by pdurbin** [comment]
@4tikhonov here you go: https://demo.dataverse.org/privateurl.xhtml?token=9c7d49e0-e8c1-4ee5-9de2-b0dab50f56e7

**Comment by pdurbin** [comment]
https://demo.dataverse.org/privateurl.xhtml?token=9c7d49e0-e8c1-4ee5-9de2-b0dab50f56e7 should look like this:

![hi-slava](https://cloud.githubusercontent.com/assets/21006/24706497/ef916df4-19dd-11e7-8aa7-cb55c899bbc0.png)

**Comment by 4tikhonov** [comment]
Yes, here it works fine. I've just tested from vagrant and problem is there. Do we need to change some settings or apply SQL for database?

**Comment by pdurbin** [comment]
@4tikhonov hmm, I've never tested Private URL from Vagrant.

**Comment by 4tikhonov** [comment]
Can you please re-open the issue? It's really important functionality for us and it doesn't work from Vagrant.

**Comment by djbrooke** [comment]
Hey @4tikhonov - I'm happy to reopen, but just so that I understand - is the issue that's it's broken in Vagrant or in your production environment (or both)?

**Comment by 4tikhonov** [comment]
Both.

**Comment by pdurbin** [comment]
@4tikhonov I'm glad you like the feature so much. 😄

**Comment by 4tikhonov** [comment]
I've tracked all steps for private URL creation and discovered that the problem was in the combination of permission bits as we're using another permissions for 'member' role to access unpublished datasets and dataverses (it can publish datasets directly). We've created the new role for people sharing private URLS (called 'viewer') and this problem was solved. 

Thank you guys for your help, it was very useful!
